src/components/ContractName.tsx

```diff
@@ -1,88 +1,30 @@
-import { BodyText } from 'components/Text'
 import { Suspense } from 'react'
 import { useSnapshot } from 'valtio'
-import EnsAddress from 'components/EnsAddress'
 import SealCredStore from 'stores/SealCredStore'
 import truncateMiddleIfNeeded from 'helpers/truncateMiddleIfNeeded'
 
 interface ContractNameProps {
-  truncate?: boolean
-  overflow?: boolean
   address: string
-  truncatedStyle?: boolean
 }
 
-interface FetchingContract extends ContractNameProps {
-  isFetching?: boolean
-}
-
-function TextBlock({
-  truncate,
-  address,
-  truncatedStyle,
-  isFetching,
-}: FetchingContract) {
-  const truncatedText =
-    truncate && address.length > 15
-      ? truncateMiddleIfNeeded(address, 11)
-      : address
-
-  return truncatedStyle ? (
-    <>
-      {isFetching && <>Fetching </>}
-      {truncatedText}
-    </>
-  ) : (
-    <BodyText small>
-      {isFetching && <>Fetching </>}
-      {truncatedText}
-    </BodyText>
-  )
-}
-
-function ContractNameComponent({
-  truncate,
-  address,
-  truncatedStyle,
-}: ContractNameProps) {
+function ContractNameSuspender({ address }: ContractNameProps) {
   const { contractNames } = useSnapshot(SealCredStore)
 
+  const contractName = contractNames[address]
+
   return (
     <>
-      {contractNames[address] ? (
-        <TextBlock
-          address={contractNames[address] || ''}
-          truncatedStyle={truncatedStyle}
-          truncate={truncate}
-        />
-      ) : (
-        <EnsAddress address={address} />
-      )}
+      {contractName
+        ? truncateMiddleIfNeeded(contractName, 17)
```

Comment:
Why do we have different lengths? It's the same component with the same width. Just do `truncateMiddleIfNeeded(contractName || address, 17)`